R/calculate_transition_disruption_metric.R

```diff
@@ -0,0 +1,121 @@
+#' Calculate the transition disruption metric, based on the IPR scenario
+#'
+#' @param data A dataset like the `*_results_portfolio.rda` outputs of PACTA.
+#' @param start_year The start year for which the TDM value should be calculated.
+#' @param ... Variables to group by.
+#'
+#' @return A tibble with the columns specified in the `...` input, as well as:
+#'   `tdm_tech`, the technology level transition disruption metric and `tdm_sec`,
+#'    the sector level transition disruption metric.
+#' @export
+#'
+#' @examples
+calculate_transition_disruption_metric <- function(data, start_year, ...) {
+  stopifnot(
+    is.data.frame(data),
+    is.numeric(start_year)
+  )
```

Comment:
Note to self: Consider asserting this is the specific kind of `data` that relates to IPR. Think about what to do if it is/not.